### ISSUE TITLE
Standardization of data not an option?

### ISSUE THREAD

**Issue by Sn0flingan** [ultralytics/yolov5#3039]
## ❔Question
When looking into the dataloaders and general inputs I notice that only normalization is done, no standardization (mean of 0 and standard deviation of 1). Is there a reason for this or is it simply a feature not yet implemented or tested?

## Additional context
Standardization of data can improve the speed at which gradient decent methods converges. When I was using ResNet I always used standardization, either on ImageNet values or on a custom set based on my dataset. But perhaps there is new information suggesting that this is not a good practice? Or perhaps yolov5 is different?

**Comment by glenn-jocher** [comment]
@Sn0flingan the batch normalization layers take care of data normalization, so there's no need to add additional manual steps for this yourself. You're free to add this step if you'd like, you simply won't see any improvement from it though.

**Comment by Sn0flingan** [comment]
@glenn-jocher Ah, but if the network is trained on centered pixels would I not need to do this when running inference as well? So I guess I should do something like img_normed = (img - img.mean()) / img.std() in my preparation of images for inference?

**Comment by glenn-jocher** [comment]
@Sn0flingan yes any preprocessing you apply in training you must also apply during inference.

**Comment by Sn0flingan** [comment]
Ah, wait, let's see if I understand this. So there are layers in the yolov5 architecture that performs standardization and when the network is exported for inference these layers are removed? 

Then we would need to re-add standardization before we do inference. But then, why does the detect.py script not contain any standardization?

**Comment by glenn-jocher** [comment]
@Sn0flingan none of the Ultralytics detection models normalize inputs. 

All input preprocessing is readily visible in the dataloaders in datasets.py:
https://github.com/ultralytics/yolov5/blob/master/utils/datasets.py

And in the Ultralytics autoShape() module when using PyTorch Hub YOLOv5 models:
https://github.com/ultralytics/yolov5/blob/b8be76f915207ef0759bfb0f1c0707c79877b763/models/common.py#L226-L250
